src/functional/swing_slide_filter.zig

```diff
@@ -174,19 +189,223 @@ pub fn compressSwing(
+    } else {
+        try appendValue(f64, current_segment.end_point.value, compressed_values);
+    }
+
+    try appendValue(usize, current_timestamp, compressed_values);
+}
+
+/// Compress `uncompressed_values` within `error_bound` using "Slide Filter" and write the
+/// result to `compressed_values`. The memory `allocator` is used to initialize the ConvexHull.
+/// If an error occurs it is returned.
+pub fn compressSlide(
+    uncompressed_values: []const f64,
+    compressed_values: *ArrayList(u8),
+    allocator: mem.Allocator,
+    error_bound: f32,
+) !void {
+
+    // Adjust the error bound to avoid exceeding it during decompression.
+    const adjusted_error_bound = if (error_bound > 0)
+        error_bound - tersets.ErrorBoundMargin
+    else
+        error_bound;
+
+    var convex_hull = try ConvexHull.init(allocator);
+    defer convex_hull.deinit();
+
+    var intercept_point: ContinousPoint = .{ .time = undefined, .value = undefined };
+
+    var upper_bound: LinearFunction = .{ .slope = undefined, .intercept = undefined };
+    var lower_bound: LinearFunction = .{ .slope = undefined, .intercept = undefined };
+    var current_segment: DiscreteSegment = .{
+        .start_point = .{ .time = 0, .value = uncompressed_values[0] },
+        .end_point = .{ .time = 1, .value = uncompressed_values[1] },
+    };
+    var new_upper_bound: LinearFunction = .{ .slope = undefined, .intercept = undefined };
+    var new_lower_bound: LinearFunction = .{ .slope = undefined, .intercept = undefined };
+
+    try convex_hull.addPoint(current_segment.start_point);
+    try convex_hull.addPoint(current_segment.end_point);
+
+    updateSlideLinearFunction(
+        current_segment,
+        &upper_bound,
+        adjusted_error_bound,
+    );
+    updateSlideLinearFunction(
+        current_segment,
+        &lower_bound,
+        -adjusted_error_bound,
+    );
+
+    var current_timestamp: usize = 2;
+    while (current_timestamp < uncompressed_values.len) : (current_timestamp += 1) {
+        // Evaluate the upper and lower bound linear functions at current timestamp.
+        const upper_limit = evaluateLinearFunctionAtTime(
+            upper_bound,
+            usize,
+            current_timestamp,
+        );
+        const lower_limit = evaluateLinearFunctionAtTime(
+            lower_bound,
+            usize,
+            current_timestamp,
         );
 
-        current_linear_function.slope = slope;
-        current_linear_function.intercept = computeInterceptCoefficient(
-            slope,
-            current_segment.start_point,
+        if ((upper_limit < (uncompressed_values[current_timestamp] - adjusted_error_bound)) or
+            (lower_limit > (uncompressed_values[current_timestamp] + adjusted_error_bound)))
+        {
+            // Recording mechanism. (the current points is outside the limits). The linear approximation
+            // crosses the interception point of the upper and lower bounds.
```

Comment:
I dont understand this comment.

Reply:
It was a comment I incorrectly added using GitHub Mobile and then deleted. However, since some of my comments are duplicated and this comment still exists it seems the delete functionality of GitHub Mobile does not work that well.